base/shell/explorer/startup.cpp

```diff
@@ -332,3 +385,85 @@ ProcessStartupItems(VOID)
 
     return res ? 0 : 101;
 }
+
+static BOOL IsSelfExplorer(VOID)
+{
+    WCHAR szPath[MAX_PATH];
+
+    GetModuleFileName(NULL, szPath, _countof(szPath));
+    PathRemoveArgsW(szPath);
+    PathRemoveBlanksW(szPath);
```

Comment:
Technically the GetModuleFileName returns a well-formed path without garbage.

Reply:
OK.